algorithm.py

```diff
@@ -39,9 +41,12 @@
     ...
 
 
-Each function returns a :py:class:`dict`, which is used to update the state of
-the current run of the algorithm. Names from the state dictionary are made
-available to downstream functions via :py:mod:`dependency_injection`. Now
+Each function returns a :py:class:`dict`, which updates the state of
```

Comment:
Why are you replacing "is used to update" with "updates"? We pass the return value of each function to `state.update`, so the passive voice is more accurate.